### PR TITLE
fix: u2o dual stack uses the corresponding svc ip

### PR DESCRIPTION
**Pull request by zcq98**
# Pull Request

- [x] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR

Examples of user facing changes:

- Bug fixes

## Which issue(s) this PR fixes

Fixes #(issue-number)

### CODE REVIEW

**Comment by zcq98** [comment]
@bobz965 Could you help review this？

**Review by bobz965 — APPROVED** [review]

**Comment by oilbeater** [comment]
@zcq98 can you give more detail description about the bug

**Comment by zcq98** [comment]
目前下发策略路由时，由于3090行match 规则的dst设置的是c.config.ServiceClusterIPRange，会导致在双栈情况下出现：
ip4.src == $subnet2019.xxx.node03_ip4 && ip4.dst == 10.233.0.0/18,fd11:1111:1112:15::/108         reroute                100.64.0.2
ip6.src == $subnet2019.xxx.node03_ip6 && ip6.dst == 10.233.0.0/18,fd11:1111:1112:15::/108         reroute            fd00:100:64::2
因此需要考虑将c.config.ServiceClusterIPRange拆分为v4和v6，达到期望效果为：
ip4.src == $subnet2019.xxx.node03_ip4 && ip4.dst == 10.233.0.0/18         reroute                100.64.0.2
ip6.src == $subnet2019.xxx.node03_ip6 && ip6.dst ==  fd11:1111:1112:15::/108         reroute            fd00:100:64::2

File: pkg/controller/subnet.go [hunk 1]
```diff
@@ -3076,18 +3076,19 @@ func (c *Controller) addPolicyRouteForU2ONoLoadBalancer(subnet *kubeovnv1.Subnet
 			klog.Error(err)
 			return err
 		}
+		v4Svc, v6Svc := util.SplitStringIP(c.config.ServiceClusterIPRange)
 		for _, cidrBlock := range strings.Split(subnet.Spec.CIDRBlock, ",") {
-			ipSuffix, nodeIP := "ip4", ip.Spec.V4IPAddress
+			ipSuffix, nodeIP, svcIP := "ip4", ip.Spec.V4IPAddress, v4Svc
```
Comment (zhangzujian) [inline pkg/controller/subnet.go:3081]:
```suggestion
			ipSuffix, nodeIP, svcCIDR := "ip4", ip.Spec.V4IPAddress, v4Svc
```

Comment (zcq98) [reply pkg/controller/subnet.go]: solved

**Review by zhangzujian — APPROVED** [review]

**Review by oilbeater — APPROVED** [review]

**Comment by zhangzujian** [comment]
We need an e2e test case for this.

**Comment by zcq98** [comment]
> We need an e2e test case for this.
I have been too busy recently, so I will gradually add e2e and unit tests this quarter.

**Pull request merged**